solr/test-framework/src/java/org/apache/solr/cloud/MiniSolrCloudCluster.java

```diff
@@ -1237,6 +1240,11 @@ public MiniSolrCloudCluster build() throws Exception {
           "solr.distributedClusterStateUpdates",
           Boolean.toString(useDistributedClusterStateUpdate));
 
+      // eager init to prevent OTEL init races caused by test setup
+      if (!disableTraceIdGeneration && TracerConfigurator.TRACE_ID_GEN_ENABLED) {
+        SimplePropagator.load();
+      }
+
```

Comment:
This part seems suspicious because you are adding this to MiniSolrCloudCluster.  Tracing isn't SolrCloud specific and isn't limited to this utility.  Maybe this should be in STCJ4 as there used to be something which you removed (not sure why)?

Reply:
I removed something that was not needed anymore https://github.com/apache/solr/pull/1841/files#diff-7cb8f3a46dbdb99ccfdcc1d07c4457167188e7366534a7e7e4a0b5aff93c1cc4L327

this is here because it's the place that boots up a multi node cluster and causing multiple OTEL inits to happen in the same jvm. I believe this is the good place for it because this is where the cluster gets setup. it is not suited for the STCJ4.

Reply:
just checking I read your message correctly. are you starting multiple solr instances as a part of a test and not using the MiniSolrCloudCluster? in this case, sure I can consider moving so it covers all solr flavors. if it's not a multi solr setup, the race does not happen and there is no issue.

Reply:
Oh I see.  Well there are number of tests manually working with multiple JettySolrRunner instances but they don't count?  Is it because we need only concern ourselves with tests touching tracing?

Reply:
> Is it because we need only concern ourselves with tests touching tracing?

no, with the "always on trace id" all tests are using tracing.

the core issue is with OTEL init method. it will not allow setting the global object more than once and will fail loudly if you try, so we'd see a test failure. I am assuming the other tests do not have the same level of concurrency on multiple solr instances coming online, so maybe the race is less likely to occur as in the case of the MiniSolrCloudCluster.

Reply:
Oh this is a race... initing OTEL _concurrently_ ?  Then I agree MiniSolrCloudCluster is probably the only way this would happen.  I thought you meant merely multiple times (e.g. in sequence).

Any way, a code comment explaining this more would help but regardless I'm +1